sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_encode.py

```diff
@@ -765,14 +765,19 @@ def encode_filter_set(value):
         else:
             if isinstance(name, str):
                 name = name.encode("utf-8")  # type: ignore
-            try:
-                descriptor, filter_value = data
-                described_filter = {
-                    TYPE: AMQPTypes.described,
-                    VALUE: ({TYPE: AMQPTypes.symbol, VALUE: descriptor}, filter_value),
-                }
-            except ValueError:
-                described_filter = data
+            if isinstance(data, (str, bytes)):
```

Comment:
can the if and else blocks be flipped here? since this block seems more like a catch all scenario and invalid filter types will be passed to the service, which returns a service error.
```
if isinstance(data, (tuple, list)):
    try:
        descriptor, filter_value = ...
    except ValueError:   # lists, tuples w/ len > 2
        ....
else:
    described_filter = data
```